doc/proposals/shared-memory-transport/sony/ms1_rtps_shared_memory_transport.md

```diff
@@ -0,0 +1,123 @@
+# Overview
+
+DDSi-RTPS(Real-Time Publish-Subscribe) is the **protocol** that enable to support vendor implementation. And RTPS specifies implementaion and message packet format and other stuff as well. On this proposal, it will descibe how we could integrate **shared memory as transport** for RTPS layer based on.
+
+# Objective
+
+Implement a shared Memory transport using the current Fast-RTPS transport API & Architecture: The transport will be used as any other transport (including serialization and RTPS encapsulation). This approach is used in other DDS implementations such as commercial release for OpenSplice and RTI Connext DDS. This is mostly related to ROS2 RMW transport, but it can be used in any DDS application or product.
+
+# Requirements
+
+- Multi-Platform support (Windows, Linux and MacOS) along with ROS2.
+- Design and Architecture Description
+- Source Code [Fast-RTPS](https://github.com/eProsima/Fast-RTPS)
+- Documents for user manual and tutorials.
+- ROS2/RMW compatible interface (do not break ROS2 userland)
+- Latency/Bandwidth improvement results.
+- Intra/Inter-Process Communication
+
+# Architecture / Design
+
+## Interoperability
+
+There is no gurantee that the whole system is constructed with eProsima Fast-RTPS, so multiple DDS implementaions may be used in a distributed system.
+To keep the interoperability between different vendor DDS implementations, the shared memory transport feature has to be implemented with the following:
+
+- All communications must take place using RTPS Messages.
+- All implementations must implement the RTPS Message Receiver.
+- Simple Participant and Endpoint Discovery Protocols.
+
+## Discovery
+
+Participant and Endpoint Discovery(PDP, EDP such as metatrrafic transmittion) will be kept as is in the current code, since this negotiation does not require shared memory use cases (all of the RTPS protocol stays put).
+
+## Memory Management
+
+- Dynamic Configuration
+  Shared memory transport should be dynamically configured when necessary, that is, when the reader exists on the same host system than the writer. This is mandatory to reduce unnecessary message passing via shared memory.
+
+  This can be detected using the GuidPrefix_t information in the locator, which includes vendor id, host id, process id and participant id.
+
+  ```
+  struct RTPS_DllAPI GuidPrefix_t
+  ```
+
+- Lifecycle
+  Shared memory lifecycle is managed by the writer corresponding to HistoryCache more likely QoS setting.
```

Comment:
'more likely' -> did you mean 'specially'? Is hard to get the intended meaning

Reply:
i believe that

> Shared memory lifecycle is managed by the writer corresponding to HistoryCache

should be good enough.